include/libdnf/base/base.hpp

```diff
@@ -95,19 +95,18 @@ class Base {
     /// Gets base variables. They can be used in configuration files. Syntax in the config - ${var_name} or $var_name.
     VarsWeakPtr get_vars() { return VarsWeakPtr(&vars, &vars_gurad); }
 
-    /// @return The system state object.
-    /// @since 5.0
-    // TODO(lukash) if it stays on public API, use WeakPtr
-    libdnf::system::State & get_system_state() { return *system_state; }
-
     void add_plugin(plugin::IPlugin & iplugin_instance);
     void load_plugins();
     plugin::Plugins & get_plugins() { return plugins; }
 
     libdnf::BaseWeakPtr get_weak_ptr() { return BaseWeakPtr(this, &base_guard); }
 
+    class Impl;
```

Comment:
It is not an issue, but the forward declaration in other header file is just above `ImplPtr<Impl> p_impl;`. may be it would be better to do it by the same approach. What do you think?

Reply:
This is a limitation of the workaround I've used to access the private member in unit tests, the `Impl` must be declared public...